.github/workflows/ci.yml

```diff
@@ -3,31 +3,51 @@ name: CI
 on: [push, pull_request]
 
 jobs:
-  ci:
+  nightly:
     runs-on: ubuntu-latest
     strategy:
       fail-fast: false
       matrix:
-        rust-toolchain: [nightly]
+        features: ["", "--all-features"]
         targets: [x86_64-unknown-linux-gnu, x86_64-unknown-none, riscv64gc-unknown-none-elf, aarch64-unknown-none-softfloat]
     steps:
     - uses: actions/checkout@v4
     - uses: dtolnay/rust-toolchain@nightly
       with:
-        toolchain: ${{ matrix.rust-toolchain }}
         components: rust-src, clippy, rustfmt
         targets: ${{ matrix.targets }}
     - name: Check rust version
       run: rustc --version --verbose
     - name: Check code format
       run: cargo fmt --all -- --check
     - name: Clippy
-      run: cargo clippy --target ${{ matrix.targets }} --all-features -- -A clippy::new_without_default
+      run: cargo clippy ${{ matrix.features }} -- -A clippy::new_without_default -Dwarnings
```

Comment:
Why remove `--target` here? In this case, `targets` specified under `dtolnay/rust-toolchain@nightly` will have no effect.

Reply:
Without `--target`, all jobs will use the default target (`x86_64-unknown-linux-gnu`) and be unable to test builds under different architectures.

Reply:
Thanks for the hint! Indeed --target should be added. I'll add it back now.